### PR TITLE
docs: add RN videos and discoverability improvements.

### PR DESCRIPTION
**Pull request by thorwebdev**
## What kind of change does this PR introduce?

- Add latest videos to docs
- Improve discoverability in navigation

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-magic-link.mdx [hunk 1]
```diff
@@ -46,6 +46,26 @@ async function signInWithEmail() {
 }
 ```
 
+</TabPanel>
+<TabPanel id="react-native" label="Expo React Native">
+Follow the [Expo Quickstart Guide](/docs/guides/getting-started/tutorials/with-expo-react-native#initialize-a-react-native-app) to corectly initialize `supabase-js` for usage with React Native.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/auth-magic-link.mdx:51]:
```suggestion
Follow the [Expo Quickstart Guide](/docs/guides/getting-started/tutorials/with-expo-react-native#initialize-a-react-native-app) to initialize `supabase-js` with React Native.
```

File: apps/docs/pages/guides/auth/auth-magic-link.mdx [hunk 2]
```diff
@@ -46,6 +46,26 @@ async function signInWithEmail() {
...
+
+When your user signs in, call [signInWithOtp()](/docs/reference/javascript/auth-signinwithotp) with their email address:
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/auth-magic-link.mdx:53]: this is a bit confusing.

"when they sign in call signInWithOtp". So they have to sign in twice?

File: apps/docs/pages/guides/auth/auth-magic-link.mdx [hunk 3]
```diff
@@ -46,6 +46,26 @@ async function signInWithEmail() {
...
+
+const { error } = await supabase.auth.signInWithOtp({
+  email: "example@email.com",
+  options: {
+    emailRedirectTo: redirectTo,
+  },
+});
+```
+
+For magic links you will need to handle deep linking into your application. Read the [docs](/docs/guides/auth/native-mobile-deep-linking).
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/auth-magic-link.mdx:68]:
```suggestion
Read the [Deep Linking Documentation](/docs/guides/auth/native-mobile-deep-linking) to learn how to handle deep linking.
```

**Review by kiwicopple — APPROVED** [review]

File: apps/docs/pages/guides/auth/auth-magic-link.mdx (hunk 2 above)
Comment (thorwebdev) [inline apps/docs/pages/guides/auth/auth-magic-link.mdx:53]:
```suggestion
To start the sign in process call [signInWithOtp()](/docs/reference/javascript/auth-signinwithotp) with their email address:
```

**Pull request merged**